foolscap/meta_data/tag_history.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime
+
+from foolscap.meta_data.io import load_tag_history
+from foolscap.meta_data.io import save_tag_history
+
+
+TAG_HISTORY = 50
+RECORD_DELETION = "{{{tag}}} -remove-from- {note} {date}"
+RECORD_ADDITION = "{{{tag}}} +added+to+ {note} {date}"
+
+
+class TagsHistory(list):
+
+    def __init__(self):
+        self.extend(load_tag_history())
+
+    def _deprecate_lines(self, delta):
+        """Determines the amount of history to deprecate."""
+        lines = 0
+        if len(self) + delta > TAG_HISTORY:
+            lines = len(self) + delta - TAG_HISTORY
+        return lines
+
+    def deprecate(self, delta):
+        """Set number of lines in history to deprecate."""
+        self.deprecate_lines = self._deprecate_lines(delta)
+
+    def save(self, lines):
+        save_tag_history(self, self.deprecate_lines)
+
+
+def format_new_history(note, new_history, style):
+    """Style the format of each tag history entry.
+
+    :param list[str] new_history: entries to be added to log.
+    :param str note: name of note being changed.
+    :param str style: format string for history entries.
+    """
+    now = datetime.now().strftime("%Y-%m-%d")
+    return [
+        style.format(tag=new, note=note, date=now)
+        for new in new_history
+    ]
+
+
+def record_tags(note, deleted, added):
+    """Record changes to tags history.
+
+    :param str note: name of note being changes.
+    :param list deleted: a list of deleted tags.
+    :param list added: a list of added tags.
+    """
+    history = TagsHistory()
+
+    deletions = format_new_history(note, deleted, RECORD_DELETION)
+    additions = format_new_history(note, added, RECORD_ADDITION)
```

Comment:
the order of `format_new_history` should be in the order of what is less likely to vary -> more likely. So that it should be:

```python
format_new_history(RECORD_DELETION, note, deleted)
format_new_history(RECORD_ADDITION, note, added)
```